src/Server.php

```diff
@@ -240,6 +242,9 @@ function ($response) use ($that, $conn, $request) {
                 $that->handleResponse($conn, $response, $request->getProtocolVersion());
             },
             function ($ex) use ($that, $conn) {
+                if (!$ex instanceof \Exception) {
+                    $ex = new \InvalidArgumentException('Unknown rejection type');
```

Comment:
This exception message is very generic. Suggestion:

```php
$ex = new \InvalidArgumentException(
     sprintf(
        'The response callback returned a rejected promise with a reason of type "%s" instead of a \Throwable or \Exception instance.'
        is_object($ex) ? get_class($ex) : gettype($ex)
    )
);
```